modules/rules_python/0.4.0/patches/propagate_pip_install_dependencies.patch

```diff
@@ -0,0 +1,78 @@
+commit 79d922cd4a4eb8f700dbdcb390cd383305ea3265
+Author: Yun Peng <pcloudy@google.com>
+Date:   Wed Sep 29 14:24:28 2021 +0200
+
+    Propagate pip_install_dependencies through module extension
+
+diff --git a/MODULE.bazel b/MODULE.bazel
+new file mode 100644
+index 0000000..2972e52
+--- /dev/null
++++ b/MODULE.bazel
+@@ -0,0 +1,17 @@
++module(
++  name="rules_python",
++  version="0.4.0",
++)
++
++pip_install = use_extension("@rules_python//bzlmod:extensions.bzl", "pip_install")
++
++pip_install.pip_install_dependencies()
++
++use_repo(pip_install,
++  "pypi__click",
++  "pypi__pip",
++  "pypi__pip_tools",
++  "pypi__pkginfo",
++  "pypi__setuptools",
++  "pypi__wheel",
++)
+diff --git a/bzlmod/BUILD b/bzlmod/BUILD
+new file mode 100644
+index 0000000..139597f
+--- /dev/null
++++ b/bzlmod/BUILD
+@@ -0,0 +1,2 @@
++
++
+diff --git a/bzlmod/extensions.bzl b/bzlmod/extensions.bzl
+new file mode 100644
+index 0000000..90993e2
+--- /dev/null
++++ b/bzlmod/extensions.bzl
+@@ -0,0 +1,11 @@
++load("@rules_python//python/pip_install:repositories.bzl", "pip_install_dependencies")
++
++def _pip_install_impl(ctx):
++  pip_install_dependencies()
++
++pip_install_dependencies_tag = tag_class()
```

Comment:
we actually don't need no-op tags here if all we want to do is to always install those dependencies. So you can actually get rid of this tag class (and line 53 altogether)

Reply:
Oh, do you mean as long as `pip_install = use_extension("@rules_python//bzlmod:extensions.bzl", "pip_install")` is called, the implementation function will be executed?

Reply:
yes, using an extension is enough to trigger its impl function.

Reply:
Thanks, done!